ksp_plugin/vessel.cpp

```diff
@@ -575,6 +591,7 @@ void Vessel::SwapPrognostication(
 }
 
 void Vessel::AppendToVesselTrajectory(
+    std::optional<Instant> const& fork_time,
```

Comment:
We pass a `DiscreteTrajectory`, so we can compute this ourselves instead of making the caller do it.

Remove the parameter, and let
```C++
bool const after_fork_time =
    !at_end_of_part_trajectory && (trajectory.is_root() || first_time > trajectory().Fork()->time);
```
(or maybe hoist it as an optional out of the loop, but given that the ancestry is always shallow here this is probably not useful).

Reply:
Done.  `is_root` is cheap, `Fork` is not.